### PR TITLE
#176454677 add in review for vendor update

### PR DESCRIPTION
**Pull request by tobieeniafe**
#### What does this PR do?
Add in review for vendor update

#### Description of Task to be completed?
Add in review for vendor update

#### How should this be manually tested?
`PUT /api/v1/user/vendor/update`

#### What are the relevant pivotal tracker stories?
#176454677

### CODE REVIEW

File: server/services/user.service.js [hunk 1]
```diff
@@ -480,25 +480,68 @@ export const verifyVendor = async ({ vendorId, adminId }) => {
 };
 
 export const updateVendor = async ({ updatedVendor, vendorId }) => {
-  const user = await getUserById(vendorId);
-  if (!user) {
-    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Vendor not found');
-  }
+  // const steps = [];
+
+  const stepToReview = {
```
Comment (andela-hpopoola) [inline server/services/user.service.js:485]: This method is too long, so you might want to create a custom function for step review to make it more readable.

```suggestion
  const stepToReview = getStepsReadyForReview(updatedVendor);
```
All of the logic to get the steps should be defined in the function.

```js
const getStepsReadyForReview = (vendor) => {
...
}

```

File: server/services/user.service.js [hunk 2]
```diff
@@ -480,25 +480,68 @@ export const verifyVendor = async ({ vendorId, adminId }) => {
...
+    stepToReview.verification.documentUpload = {
+      status: VENDOR_INFO_STATUS.IN_REVIEW,
+    };
   }
+
+  if (
+    updatedVendor.phone ||
+    updatedVendor.phone2 ||
+    updatedVendor.address ||
+    (updatedVendor.vendor && updatedVendor.vendor.companyName)
```
Comment (andela-hpopoola) [inline server/services/user.service.js:521]: You will also need to check for the following fields too:

* companyLogo
* entity
* redanNumber
* socialMedia
* website

Yes, I know that some of them are not used in the current frontend, however, we don't want to wait till they are used before we do the right thing.

**Review by andela-hpopoola — CHANGES_REQUESTED** [review]
Thank you for working on this. I left a few comments, you are currently on the right track. You will need to remove the commented out code for the previous implementation and write tests to ensure that is ready to be merged. Nice work 👍🏽

File: server/services/user.service.js (hunk 1 above)
Comment (andela-hpopoola) [reply server/services/user.service.js:485]: Can you please add the step to review code in a new function as suggested above to make the code easier to read?

File: server/services/user.service.js [hunk 3]
```diff
@@ -585,7 +631,12 @@ export const removeDirector = async ({ directorId, user }) => {
   try {
     return User.findByIdAndUpdate(
       user._id,
-      { $set: { 'vendor.directors': directors } },
+      {
+        $set: {
+          'vendor.directors': directors,
+          'vendor.verification.directorInfo.status': VENDOR_INFO_STATUS.IN_REVIEW,
```
Comment (andela-hpopoola) [inline server/services/user.service.js:637]: Can we ensure that this only takes place when an admin has not been verified? If an admin has been verified, there is no need to set this to `IN REVIEW` when updating/deleting directors.

Comment (andela-hpopoola) [reply server/services/user.service.js:637]: Also, can we set the status of unverified vendor to `IN REVIEW` when a director is deleted as discussed this morning?

File: test/controllers/user.controller.test.js [hunk 4]
```diff
@@ -1796,6 +1792,12 @@ describe('User Controller', () => {
               expect(res.body.user.vendor.taxCertificate).to.be.eql(data.vendor.taxCertificate);
               expect(res.body.user.vendor.socialMedia.length).to.be.eql(2);
               expect(res.body.user.vendor.directors.length).to.be.eql(2);
+              expect(res.body.user.vendor.verification.companyInfo.status).to.be.eql('In Review');
```
Comment (andela-hpopoola) [inline test/controllers/user.controller.test.js:1795]: We will need a different test to verify this. You might also need to add `phone` to the data object (on line 1730) so that we can test that it is also updated.

https://github.com/Highrachy/ballers-api/blob/66fd1c6072febf7127efcf3638cdcd4945da1ef8/test/controllers/user.controller.test.js#L1730

**Review by andela-hpopoola — CHANGES_REQUESTED** [review]
Thank you for working on this. I left a few comments on ensuring that we cover all edge cases.

File: test/controllers/user.controller.test.js [hunk 5]
```diff
@@ -1917,6 +1918,25 @@ describe('User Controller', () => {
...
+            });
+        });
+      });
+
+      context('when a valid token is used & vendor is not verified', () => {
+        beforeEach(async () => {
+          await User.findByIdAndUpdate(vendorUser._id, { 'vendor.verified': false });
+        });
+
+        it('returns deletes director', (done) => {
```
Comment (andela-hpopoola) [inline test/controllers/user.controller.test.js:1932]:
```suggestion
	        it('sets director verification to `In Review`', (done) => {
```

File: test/controllers/user.controller.test.js [hunk 6]
```diff
@@ -1917,6 +1918,25 @@ describe('User Controller', () => {
               expect(res.body.user.vendor.directors.length).to.be.eql(2);
               expect(res.body.user.vendor.directors[0]._id).to.not.eql(nonSignatoryId);
               expect(res.body.user.vendor.directors[1]._id).to.not.eql(nonSignatoryId);
+              expect(res.body.user.vendor.verification.directorInfo.status).to.be.eql('Pending');
```
Comment (andela-hpopoola) [inline test/controllers/user.controller.test.js:1921]: Can we set the default of this to `verified` so that this returns `Verified`. Having it as `Pending` feels weird for a verified vendor (not a major blocker, it is just to ensure that the future reading of the test is easily understood).

**Review by andela-hpopoola — APPROVED** [review]
Thank you for working on this. I added 2 non-blocking comments, so feel free to merge when you have resolved them. Nice work.
